### PR TITLE
Add bring your own route table support

### PR DESCRIPTION
**Pull request by gavinfish**
---

- When one of subnets associate route table while the other not, error
- When two subnets associate with the same route table, error
- Subnets with different route tables should be validate now

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
spring cloud

File: src/spring-cloud/azext_spring_cloud/_validators.py [hunk 1]
```diff
@@ -420,3 +418,21 @@ def _validate_resource_group_name(name, message_name):
...
+        if subnet.id.lower() == namespace.service_runtime_subnet.lower() and subnet.route_table:
+            runtime_route_table_id = subnet.route_table.id
+
+    if app_route_table_id != "" and runtime_route_table_id != "":
+        if app_route_table_id == runtime_route_table_id:
+            raise CLIError('--service-runtime-subnet and --app-subnet should associate with different route tables.')
+    if app_route_table_id != "" and runtime_route_table_id == "" \
+            or app_route_table_id == "" and runtime_route_table_id != "":
+        raise CLIError(
+            '--service-runtime-subnet and --app-subnet should both associate with different route tables or neither.')
```
Comment (yungezz) [inline src/spring-cloud/azext_spring_cloud/_validators.py:438]: confusing...

File: src/spring-cloud/azext_spring_cloud/_validators.py [hunk 2]
```diff
@@ -420,3 +418,21 @@ def _validate_resource_group_name(name, message_name):
...
     if matchObj is None:
         raise CLIError('--{0} must conform to the following pattern: \'^[-\\w\\._\\(\\)]+$\'.'.format(message_name))
+
+
+def _validate_route_table(namespace, vnet_obj):
+    app_route_table_id = ""
+    runtime_route_table_id = ""
+    for subnet in vnet_obj.subnets:
+        if subnet.id.lower() == namespace.app_subnet.lower() and subnet.route_table:
+            app_route_table_id = subnet.route_table.id
```
Comment (ShichaoQiu) [inline src/spring-cloud/azext_spring_cloud/_validators.py:428]: Not sure if the subnet.route_table.id will be None sometimes?

File: src/spring-cloud/azext_spring_cloud/_validators.py [hunk 3]
```diff
@@ -420,3 +418,21 @@ def _validate_resource_group_name(name, message_name):
...
+        if subnet.id.lower() == namespace.service_runtime_subnet.lower() and subnet.route_table:
+            runtime_route_table_id = subnet.route_table.id
+
+    if app_route_table_id != "" and runtime_route_table_id != "":
```
Comment (ShichaoQiu) [inline src/spring-cloud/azext_spring_cloud/_validators.py:432]: How about `if app_route_table_id  and runtime_route_table_id`?

File: src/spring-cloud/azext_spring_cloud/_validators.py (hunk 1 above)
Comment (gavinfish) [reply src/spring-cloud/azext_spring_cloud/_validators.py:438]: The two subnets should both bind with different route tables or neither.

**Review by ShichaoQiu — APPROVED** [review]

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**